server.js

```diff
@@ -0,0 +1,47 @@
+const app = require("./backend/app");
+const debug = require("debug")("node-angular");
+const http = require("http");
+
+const normalizePort = val => {
+    const port = parseInt(val, 10);
+    if (isNaN(port)) {
+        return val;
+    }
+    if (port>=0) {
+       return port;
+    }
+    return false;
+}
+
+const onError = error => {
+    if (error.syscall !== 'listen') {
+        throw error;
+    }
+
+    const bind = typeof addr === "string" ? "pipe" + addr: "port" + port;
+    switch (error.code) {
+        case "EACCES":
+            console.error(bind + "requires elevated privilages");
+            process.exit(1);
+            break;
+        case "EADDRINUSE":
+            console.error(bind + "is already in use");
+            process.exit(1);
+            break;
+        default:
+            throw error;
+    }
+ };
+ const onListening = () => {
+    const addr = server.address();
+    const bind = typeof addr === "string" ? "pipe" + addr: "port " + port;
+    debug('Listening on ' + bind);
+ }
+
+const port = normalizePort(process.env.PORT || "3000");
```

Comment:
Could use `parseInt(process.env.PORT) || 3000` that will fallback to 3000 if parseInt will fail (error/undefined/null/NaN)